common/flatpak-dir.c

```diff
@@ -7594,31 +7613,76 @@ flatpak_dir_install (FlatpakDir          *self,
         }
       else
         {
+          /* If we find bindfs on the system, we try to make a pull using the bindfs mount otherwise fallback.
+             Also, we need a separate OSTree pointer to bindfs repo, local to this block.
+             Reason being that if we assign the bindfs repo to `child_repo` which has life outside
+             this block, while deploying, it will lead to open FDs and unmounting the bindfs mount
+             will fail with EBUSY (inside system-helper's handle_deploy()). */
```

Comment:
hmm, this is interesting. Earlier I have tried out :
``` 
child_repo = g_steal_pointer (&bindfs_repo)  
...
g_object_unref (child_repo) // to close it
``` 
I was error-ing out for cleanup code path:
```
#0  0x00007ffff77263c5 in  () at /lib64/libglib-2.0.so.0
#1  0x00007ffff77276fd in g_logv () at /lib64/libglib-2.0.so.0
#2  0x00007ffff77278d3 in g_log () at /lib64/libglib-2.0.so.0
#3  0x000000000045f544 in glib_autoptr_cleanup_OstreeRepo (_ptr=<synthetic pointer>)
    at /usr/include/ostree-1/ostree-autocleanups.h:55
#4  0x000000000045f544 in flatpak_dir_install
    (self=0x4f1800, no_pull=<optimized out>, no_deploy=0, no_static_deltas=<optimized out>, reinstall=0, state=state@entry=0x501b10, ref=0x630790 "app/org.gnome.Photos/x86_64/stable", opt_commit=0x503970 "a89502fbf33367163e5e09d785284063658577babab1705d088dca5dea3c9e14", opt_subpaths=<optimized out>, progress=0x7fffe400dc40, cancellable=0x0, error=0x7fffffffd3e8)
    at common/flatpak-dir.c:7546
#5  0x0000000000470fa6 in flatpak_transaction_run
    (self=self@entry=0x4fc930, cancellable=cancellable@entry=0x0, error=error@entry=0x7fffffffd438)
    at common/flatpak-transaction.c:2384

```
But `g_clear_pointer (&child_repo)` doesn't give me any error. I wonder why? :thinking: 

Reducing the scope might work but it will introduce changes in the upper `else if` blocks and need to be declared in those blocks separately.

Reply:
Why? Because calling `g_object_unref (child_repo)` does not set `child_repo = NULL`, so when you leave the scope, `g_object_unref (child_repo)` is called again ⇒ double free.